validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -699,10 +700,11 @@ def _attempt_to_peer_with_endpoint(self, endpoint):
             # if the connection uri wasn't found in the network's
             # connections, it raises a KeyError and we need to add
             # a new outbound connection
-            self._temp_endpoints[endpoint] = EndpointInfo(
-                EndpointStatus.PEERING,
-                time.time(),
-                INITIAL_RETRY_FREQUENCY)
+            with self._lock:
```

Comment:
Which lock is this?  What else is it guarding?

Reply:
This is the lock is in ConnectionManager, it is guarding self._temp_endpoints but is also used around other things. I saw a "dictionary change size during iteration error " in _check_temp_endpoints which is already wrapped in a lock.

Reply:
Do you have other suggestions about what may cause this error or how to fix it?

Reply:
No, I think it's the right call.  I think the lock should be renamed to its specific use.  `_lock` is too generic, and it's only guarding the temp connections collection.